File: library/src/test/java/io/fotoapparat/test/ImmediateExecutorTest.java
package io.fotoapparat.test;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class ImmediateExecutorTest {

    @Mock
    Runnable runnable;

    ImmediateExecutor testee = new ImmediateExecutor();

    @Test
    public void execute() throws Exception {
        // When
        testee.execute(runnable);

        // Then
        verify(runnable).run();
    }

}

File: fotoapparat/src/test/java/io/fotoapparat/test/ImmediateExecutorTest.kt
package io.fotoapparat.test

import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class ImmediateExecutorTest {

    @Mock
    lateinit var runnable: Runnable

    @Test
    fun execute() {
        // When
        ImmediateExecutor.execute(runnable)

        // Then
        verify(runnable).run()
    }

}